### PR TITLE
Add separate SPIFFE integeration test build target.

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

Commit Message: Add separate SPIFFE integeration test build target.
Additional Description: SPIFFE certificate validator is not a "core" extension, so this commit separates out its integration test from ssl_integration_test in order to remove the dependency of core integration tests on non-core SPIFFE cert validator extension.

resolves: https://github.com/envoyproxy/envoy/pull/14884#discussion_r587886638 
ref: https://github.com/envoyproxy/envoy/issues/9953

### CODE REVIEW

**Comment by antoniovicente** [comment]
Thanks for the help with this.  I'll take a look a bit later.

Also, I think that test/extensions/transport_sockets/tls/cert_validator/factory_test.cc will fail if the SPIFFE cert verifier extension were disabled.

**Comment by mathetake** [comment]
> Also, I think that test/extensions/transport_sockets/tls/cert_validator/factory_test.cc will fail if the SPIFFE cert verifier extension were disabled.

I think It's not. It just uses the string literal "spiffe" and that's it.

File: test/extensions/transport_sockets/tls/integration/BUILD [hunk 1]
```diff
@@ -41,3 +39,19 @@ envoy_cc_test(
         "@envoy_api//envoy/extensions/transport_sockets/tls/v3:pkg_cc_proto",
     ],
 )
+
+envoy_cc_test(
```
Comment (lizan) [inline test/extensions/transport_sockets/tls/integration/BUILD:43]: shall we move this to envoy_extension_cc_test in corresponding cert validator extension directory?

Comment (mathetake) [reply test/extensions/transport_sockets/tls/integration/BUILD]: Thanks!

**Review by antoniovicente — APPROVED** [review]
Thanks for moving the spiffe test to the cert validator directory.

**Pull request merged**